testsuite/tests/printing-types/pprint_polyvariant.ml

```diff
@@ -0,0 +1,17 @@
+(* TEST
+   * bytecode
+   include ocamlcommon
+*)
+
+let pa s =
+  let lb = Lexing.from_string s in
+  Parse.implementation lb
+;;
+let pr st =
+  Pprintast.string_of_structure st
+;;
+
+{| type t = [ | w ] |} |> pa |> pr |> print_string ;; print_newline () ;;
+{| type t = [ | `A of int ] |} |> pa |> pr |> print_string ;; print_newline () ;;
+{| type t = [ `A of int ] |} |> pa |> pr |> print_string ;; print_newline () ;;
+
```

Comment:
There is in fact a parsing round-trip test in `parsing/source.ml`. I think it would be better to move your test here.